src/Symfony/Component/JsonPath/Tests/Test/JsonPathAssertionsTraitTest.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Symfony\Component\JsonPath\Tests\Test;
+
+use PHPUnit\Framework\AssertionFailedError;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\JsonPath\Test\JsonPathAssertionsTrait;
+
+class JsonPathAssertionsTraitTest extends TestCase
+{
+    use JsonPathAssertionsTrait;
+
+    public function testAssertJsonPathEqualsOk()
+    {
+        self::assertJsonPathEquals([1], '$.a[2]', self::getSimpleCollectionCrawlerData());
+    }
+
+    public function testAssertJsonPathEqualsOkWithTypeCoercion()
+    {
+        self::assertJsonPathEquals(['1'], '$.a[2]', self::getSimpleCollectionCrawlerData());
+    }
+
+    public function testAssertJsonPathEqualsKo()
+    {
+        try {
+            self::assertJsonPathEquals([2], '$.a[2]', self::getSimpleCollectionCrawlerData());
+
+            $this->fail();
```

Comment:
this won't work fine, because the failure in `$this->fail()` will also be caught by your catch block.